protected_tags.go

```diff
@@ -0,0 +1,141 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// ProtectedTagsService handles communication with the protected tag methods of the GitLab API.
```

Comment:
Please try to keep comments below 80 chars. Some doc URLs are longer and that is fine, but normal comments should not exceed 80 chars.